### PR TITLE
fix editor to do

### PR DESCRIPTION
**Pull request by bofeng-song**

### CODE REVIEW

File: native/cocos/core/assets/EffectAsset.cpp [hunk 1]
```diff
@@ -134,9 +137,15 @@ EffectAsset *EffectAsset::get(const ccstd::string &name) {
 void EffectAsset::onLoaded() {
     ProgramLib::getInstance()->registerEffect(this);
     EffectAsset::registerAsset(this);
-    //cjh TODO:    if (!EDITOR){
-    //cjh    legacyCC.game.once(legacyCC.Game.EVENT_ENGINE_INITED, this._precompile, this);
-    // }
+#if (CC_EDITOR == 0)
```
Comment (minggo) [inline native/cocos/core/assets/EffectAsset.cpp]: I think use `!CC_EDITOR` is better.

File: native/cocos/engine/BaseEngine.h [hunk 2]
```diff
@@ -109,6 +109,8 @@ class CC_DLL BaseEngine : public CallbacksInvoker,
      @brief Get engine scheduler.
      */
     virtual SchedulerPtr getScheduler() const = 0;
+
+    virtual bool isInited() = 0;
```
Comment (minggo) [inline native/cocos/engine/BaseEngine.h]: Can add const here.

File: native/cocos/core/assets/EffectAsset.cpp (hunk 1 above)
Comment (bofeng-song) [reply native/cocos/core/assets/EffectAsset.cpp]: has been fixed

File: native/cocos/engine/BaseEngine.h (hunk 2 above)
Comment (bofeng-song) [reply native/cocos/engine/BaseEngine.h]: has been fixed

**Review by minggo — APPROVED** [review]

**Review by qiuguohua — APPROVED** [review]

**Review by dumganhar — APPROVED** [review]

**Pull request merged**